src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreUtil.cs

```diff
@@ -0,0 +1,738 @@
+            string last = "";
+            foreach (DataRow row in dt.Rows)
+            {
+                string dbName = Convert.ToString(row["DatabaseName"], System.Globalization.CultureInfo.InvariantCulture);
+                if (!this.excludedDB.Contains(dbName) && !dbName.Equals(last))
+                {
+                    bool found = false;
+                    foreach (string str in databaseNames)
+                    {
+                        if (StrEqual(str, dbName))
+                        {
+                            found = true;
+                            break;
+                        }
+                    }
+                    if (found == false)
+                    {
+                        databaseNames.Add(dbName);
+                    }
+                }
+                last = dbName;
+            }
+            return databaseNames;
+        }
+        */
+
+        /// <summary>
+        /// Reads backup file header to get source database names
+        /// If valid credential name is not provided for URL throws exception while executing T-sql statement
+        /// </summary>
+        /// <param name="bkdevList">List of backup device items</param>
+        /// <param name="credential">Optional Sqlserver credential name to read backup header from URL</param>
+        public List<String> GetSourceDbNames(ICollection<BackupDeviceItem> bkdevList, string credential = null)
+        {
+            List<string> databaseNames = new List<string>();
+            foreach (BackupDeviceItem bkdev in bkdevList)
+            {
+                // use the Restore public API to do the Restore Headeronly query
+                Restore res = new Restore();
+                res.CredentialName = credential;
+                res.Devices.Add(bkdev);
+                
+                DataTable dt = res.ReadBackupHeader(this.server);
+                if (dt != null)
+                {
+                    foreach (DataRow dr in dt.Rows)
+                    {
+                        if (dr != null && !(dr["DatabaseName"] is DBNull))
+                        {
+                            string dbName = (string)dr["DatabaseName"];
+                            bool found = false;
+                            foreach (string str in databaseNames)
+                            {
+                                if (StrEqual(str, dbName))
+                                {
+                                    found = true;
+                                    break;
+                                }
+                            }
+                            if (found == false)
+                            {
+                                databaseNames.Add(dbName);
+                            }
+                        }
+                    }
+                }
+            }
+            return databaseNames;
+        }
+
+        public bool StrEqual(string strA, string strB)
```

Comment:
> public bool StrEqual [](start = 7, length = 21)

seems like this isn't a very use method in my opinion, rather just call compare directly where needed.  If this method isn't called outside this class could you make it private to not encourage it being used else where?